### PR TITLE
Add CLI command to import info.json files

### PR DESCRIPTION
**Pull request by zvecr**
## Description

As a side project, i've been working on a replacement of kbfirmware that exports `info.json` files.

Looking at what the user does next, this cli command stubs out the `qmk new-keyboard` side of things while allowing import of `info.json` files.

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ ] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Checklist

- [ ] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [ ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ ] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

**Comment by tzarc** [comment]
Sorry about that, GitHub decided to delete the `develop` branch from upstream when we merged it, despite being told otherwise. Reopened.

File: docs/cli_commands.md [hunk 1]
```diff
@@ -230,14 +230,22 @@ qmk list-keymaps -kb planck/ez
...
-This command creates a new keyboard based on available templates.
+This command creates a new keyboard based on available templates or imports an existing config.
 
 This command will prompt for input to guide you though the generation process.
 
 **Usage**:
 
 ```
-qmk new-keyboard
+qmk new-keyboard [-kb KEYBOARD] [filename]
```
Comment (Erovia) [inline docs/cli_commands.md:240]:
```suggestion
qmk new-keyboard [filename]
```

File: lib/python/qmk/cli/new/keyboard.py [hunk 2]
```diff
@@ -1,11 +1,56 @@
...
+    keymap_data = {
+        "keyboard": name,
+        "layout": layout_name,
+        "layers": [["KC_NO" for _ in range(0, layout_length)]],
+    }
+
+    return json.dumps(keymap_data)
+
+
+def import_info_json(name, info):
```
Comment (Erovia) [inline lib/python/qmk/cli/new/keyboard.py:27]:
```suggestion
def import_info_json(info):
```

File: lib/python/qmk/cli/new/keyboard.py [hunk 3]
```diff
@@ -1,11 +1,56 @@
...
+    # Validate to ensure we don't have to deal with bad data - handles stdin/file
+    info_data = hjson.load(info)
+    keyboard_validate(info_data)
+
```
Comment (Erovia) [inline lib/python/qmk/cli/new/keyboard.py:31]:
```suggestion

   try:
       name = info_data["keyboard_name"]
   except KeyError:
       cli.log.error("%s does not contain 'keyboard_name' field.", info.name)
       exit(1)
```

File: lib/python/qmk/cli/new/keyboard.py [hunk 4]
```diff
@@ -1,11 +1,56 @@
...
+    # This is the deepest folder in the expected tree
+    keyboard_keymap.parent.mkdir(parents=True, exist_ok=True)
+
+    # Dump out all those lovely files
+    keyboard_info.write_text(json.dumps(info_data))
+    keyboard_rules.write_text("")
+    keyboard_keymap.write_text(_gen_dummy_keymap(name, info_data))
 
 
+@cli.argument('-kb', '--keyboard', arg_only=True, help='Specify keyboard name. Example: 1upkeyboards/1up60hse')
```
Comment (Erovia) [inline lib/python/qmk/cli/new/keyboard.py:46]:
```suggestion
```

File: lib/python/qmk/cli/new/keyboard.py [hunk 5]
```diff
@@ -1,11 +1,56 @@
...
+@cli.argument('filename', type=qmk.path.FileType('r'), nargs='?', arg_only=True, help='info JSON file')
 @cli.subcommand('Creates a new keyboard')
 def new_keyboard(cli):
     """Creates a new keyboard
     """
-    # TODO: replace this bodge to the existing script
-    cli.run(['util/new_keyboard.sh'], capture_output=False)
+    if cli.args.filename:
+        import_info_json(cli.args.keyboard, cli.args.filename)
```
Comment (Erovia) [inline lib/python/qmk/cli/new/keyboard.py:53]:
```suggestion
        import_info_json(cli.args.filename)
```

**Review by Erovia — CHANGES_REQUESTED** [review]

**Pull request closed without merging**